### ISSUE TITLE
Sometimes models load very slowly

### ISSUE THREAD

**Issue by Jotakak-yu** [microsoft/VPTQ#125]
I honestly don't know what the exact trigger condition is, before 0.0.4, this situation usually occurs when I am not running in the vptq root directory, and this problem is usually accompanied by a slow token generation problem; Now this happens regardless of whether I'm running in the root directory or not, but when the model is loaded, the token generation speed is normal. I am very confused now and hope to get your help

**Comment by YangWang92** [comment]
May I ask if the issue is with slow inference speed or slow model loading? Also, could you share what hardware you are using for testing? Thank you!

**Comment by YangWang92** [comment]
I’m guessing you might need to run the same model repeatedly with different input or context? In that case, you probably only need to load the model once and perform inference on new inputs repeatedly to avoid reloading the model each time, similar to what’s done here https://github.com/microsoft/VPTQ?tab=readme-ov-file#python-api-example and here https://github.com/microsoft/VPTQ/blob/main/vptq/app_utils.py#L120.

**Comment by Jotakak-yu** [comment]
@YangWang92 Thanks for your reply! I'm sorry that I didn't provide enough information. I'm testing on A6000, and the problem I'm facing is slow model loading (for version 0.0.4). I just used the command "python -m vptq --model=VPTQ-community/Qwen2.5-72B-Instruct-v8-k65536-0-woft --chat" to test, so I didn't run the model repeatedly. Everything was fine for the first few runs, but then I found that it took a long time for the model to load (more than 5 minutes) after running the command, and it was impossible to get back to normal speed after that. Since it's a public server, I can't restart it. I tried to reinstall vptq or compile it from source code, but it didn't work.

**Comment by YangWang92** [comment]
> [@YangWang92](https://github.com/YangWang92) Thanks for your reply! I'm sorry that I didn't provide enough information. I'm testing on A6000, and the problem I'm facing is slow model loading (for version 0.0.4). I just used the command "python -m vptq --model=VPTQ-community/Qwen2.5-72B-Instruct-v8-k65536-0-woft --chat" to test, so I didn't run the model repeatedly. Everything was fine for the first few runs, but then I found that it took a long time for the model to load (more than 5 minutes) after running the command, and it was impossible to get back to normal speed after that. Since it's a public server, I can't restart it. I tried to reinstall vptq or compile it from source code, but it didn't work.


Thanks for your reply~
Let me check it. I found a similar problem when I used the virtual machine from a public cloud service provider. It always takes 5 minutes to load the model (e.g., 72B, >18GB) from disk to GPU memory. Can you check the io bandwidth using the `iotop` command? Is it more related to disk performance or PCIe bandwidth? 

BTW, can you load an standard/open source model (e.g., qwen2.5-32B or 14B, ~20GB) by hugging the transformer library from disk to GPU memory? How long does it take?
```bash
from transformers import AutoModelForCausalLM, AutoTokenizer

model_name = "Qwen/Qwen2.5-14B-Instruct"

model = AutoModelForCausalLM.from_pretrained(
    model_name,
    torch_dtype="auto",
    device_map="auto"
)
tokenizer = AutoTokenizer.from_pretrained(model_name)
```

**Comment by Jotakak-yu** [comment]
I apologize for the late reply. I checked the disk I/O during model loading (however, the I/O information of the VPTQ loading process does not seem to be reflected in iotop; in the subsequent tests on two servers, the read speed during loading was 0). I noticed that the process of "Replacing linear layers" is very fast (1913.24it/s), which means that once a step begins, the loading speed is not an issue. However, there is a waiting period before each step starts. During this waiting period, no data is actually transferred, making it feel like it's stuck. After "Replacing linear layers" is completed, it takes several minutes before loading into GPU memory begins.

When testing on another server where VPTQ performs well, I found that after "Replacing linear layers," there is a process of "Fetching xx files," which does not occur on the problematic server.

The strange thing about this issue is that it appears without any apparent reason (at least to me). One moment the startup can load normally, and the next moment, each stage of the loading process starts with a long delay.

Additionally, I tested loading Qwen2.5-14B-Instruct via the Hugging Transformer library according to your suggestion, and it took 9 seconds. It seems that the problem does not originate from disk performance or PCIe bandwidth but rather from the waiting period before each stage of the loading process begins.
